### DISCUSSION TITLE
spacy.training doesn't get recognized

### DISCUSSION THREAD

**Discussion by saguillo2000** [explosion/spaCy#7070] [Help: Other Questions] [answered]
## How to reproduce the problem
 I tried to reinstall all, but the problem is that the package training is installed in the spacy package, and I don't know why it isn't recognized. I tried also Google Collab and displays the same issue:

![image](https://user-images.githubusercontent.com/47833532/107922512-66675380-6f70-11eb-8e72-f81f210b26df.png)

```bash
# Cannot find reference 'training' in '__init__.py | __init__.py'
```

## PyCharm

* Operating System: Windows10
* Python Version Used: 3.8
* spaCy Version Used: 3.0.3
* Environment Information: Using venv

**Comment by svlandeg** [comment, marked as answer]
The line in the screenshot isn't finished, because it says `from spacy.training` but it should be followed with an `import ...` to specify what exactly you want to import.

What happens when you try this?
```
from spacy.training import Example
```

**Reply by saguillo2000** [reply]
When I try the Example it displays me that doesn't recognize the package and is not installed:
![image](https://user-images.githubusercontent.com/47833532/107923772-3faa1c80-6f72-11eb-9d46-a88a9e7c9d49.png)
When I press to do an installation displays me the following message:
![image](https://user-images.githubusercontent.com/47833532/107923841-58b2cd80-6f72-11eb-9347-58fa9ccafbcb.png)

**Reply by svlandeg** [reply]
Hm. `training` is a new module since 3.0, so I wonder whether perhaps a previous installation of 2.x is obscuring things here? Any chance you could run a clean install from scratch, in a clean venv?

**Reply by saguillo2000** [reply]
it gives me the same problem :').  I used the following configurations for the new project:
![image](https://user-images.githubusercontent.com/47833532/107932707-27d89580-6f7e-11eb-817e-656e9b764560.png)
And it gives me the same issue. For creating the venv I used the classic commands, python -m venv venv and activated it witg venv\Scripts\activate.bat.
I tried also in the Google Collab and give me the same issue. If it is easier to install spacy in Ubuntu may be I should try to get a VM.
In your case what OS and how did the installation? 
PD: The folder of training is in the spacy package but I don't know why Pycharm doesn't recognize it

**Reply by svlandeg** [reply]
What if you create a new virtual environment with `python -m venv`, somewhere on your disk, install `spacy` there from scratch, and then point to that environment and the python executable inside it, when selecting your interpreter in pycharm?

Something like this:
![image](https://user-images.githubusercontent.com/8796347/107937828-c7992200-6f84-11eb-8765-0448bf9cdf68.png)

Because I worry about that the combination between `conda` and `venv` is resulting in some sort of conflict?

**Reply by saguillo2000** [reply]
I think you answered the problem, but few questions. Now I have installed in the new interpreter spacy as I you see in the following photo:
![image](https://user-images.githubusercontent.com/47833532/107941730-71c77880-6f8a-11eb-8f0f-4f2638fe871c.png)
Even if I'm using the interpreter generated by the venv, I have to activate it always to use the spacy? If I activate it I think it works, but the interpreter has all the libraries installed so I don't understand why I have to activate it or may be I'm doing sthg wrong.
Really thank for the time and the solutions!!!

**Reply by svlandeg** [reply]
PyCharm works such that you define the Python interpreter once for a project - either upon creation or afterwards like in my screenshot. Now, every time you'll open up Pycharm with this specific project, it will automatically use that specific virtual environment that you selected. So, you don't need to specifically "activate" anything - PyCharm will do it for you.

That said, if you want to run any other code/script on a command line outside of Pycharm, then you'd go to your console and activate the same environment, and run whatever you want.

**Reply by svlandeg** [reply]
(and you're welcome, ofcourse! I'll assume the problem is resolved and will select this entire thread as final answer so others see that this is addressed)
